### DISCUSSION TITLE
helm-controller: action conditions represent current state or action occurence?

### DISCUSSION THREAD

**Discussion by seaneagan** [fluxcd/flux2#179] [Q&A] [answered]
Currently the status of action conditions (Installed, Upgraded, Rolledback, Uninstalled, Tested) persists for as long as the spec (generation) and calculated state (e.g. resolved chart version, values from secret/configmap, out of band helm actions) do. They don't take into account for example resetting:

1. `Installed` after uninstall remediation.
2. `Upgraded` after rollback remediation.
3. `Uninstalled` after install/upgrade retry.
4. `Rolledback` after upgrade retry.

Do these conditions represent:

a. current state of the release (more consistent with how conditions were intended to be used)
b. action occurrence/success against latest spec/state (current behavior)

None of the conditions are used by the controller itself, so it should be a question of human UX. At least `Tested` is useful. The others could potentially be refactored to `Deployed` / `Remediated` using the reason/message to specify which action led to that.

**Comment by hiddeco** [comment, marked as answer]
What we know from talking to Flux (v1) / Helm Operator users is that they like to be able to see "what happened during a release", and what resulted in the failed state. In the first release we did not have anything in place to push events, so the only place this information could be captured was the conditions field.

I do agree that recording the current state of the release is indeed more consistent with how conditions were intended to be used, as long as we broadcast an event with the result of every action that was run.

**Reply by seaneagan** [reply]
Resolved here: https://github.com/fluxcd/helm-controller/pull/75
